src/main/java/reactor/core/scheduler/ForkJoinPoolScheduler.java

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright (c) 2011-2017 Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.scheduler;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinTask;
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+
+import reactor.core.Disposable;
+import reactor.util.Logger;
+import reactor.util.Loggers;
+
+public class ForkJoinPoolScheduler implements Scheduler {
+
+	private final ForkJoinPool             pool;
+	private final ScheduledExecutorService scheduler;
+
+	public ForkJoinPoolScheduler(int parallelism, ThreadFactory threadFactory) {
```

Comment:
the constructor can also be package-protected. I'd still put javadoc on it to make a note that `threadFactory` is only applicable to the `ScheduledExecutorService` that is used internally for timer-based tasks.